velox/exec/tests/JoinFuzzer.cpp

```diff
@@ -86,55 +94,116 @@ class JoinFuzzer {
+  // columns.
+  void generateProbeInput();
 
-  /// Same as generateProbeInput() but copies over 10% of the input in the probe
-  /// columns to ensure some matches during joining. Also generates an empty
-  /// input with a 10% chance.
-  std::vector<RowVectorPtr> generateBuildInput(
-      const std::vector<RowVectorPtr>& probeInput,
-      const std::vector<std::string>& probeKeys,
-      const std::vector<std::string>& buildKeys);
+  // Same as generateProbeInput() but copies over 10% of the input in the probe
+  // columns to ensure some matches during joining. Also generates an empty
+  // input with a 10% chance.
+  void generateBuildInput();
+
+  void shuffleJoinKeys();
 
-  void shuffleJoinKeys(
-      std::vector<std::string>& probeKeys,
-      std::vector<std::string>& buildKeys);
+  void addPlansWithTableScan(std::vector<PlanWithSplits>& altPlans);
+
+  // Generates the splits for group execution plans. The function splits the
+  // inputs into a number of groups by partitioning on the join keys. It then
+  // writes the partitioned inputs into separate files to generate grouped
+  // splits.
+  void generateTableScanSplitsWithGroup();
+
+  // Splits the input into groups by partitioning on the join keys. If 'isProbe'
+  // is true, this splits the input for probe side, otherwise for build.
+  std::vector<std::vector<RowVectorPtr>> splitInputByGroup(bool isProbe);
+
+  // Generates the grouped splits. If 'isProbe' is true, this generates grouped
+  // splits for probe side, otherwise for build.
+  void generateSplitsWithGroup(bool isProbe);
 
   RowVectorPtr execute(const PlanWithSplits& plan, bool injectSpill);
 
   std::optional<MaterializedRowMultiset> computeDuckDbResult(
-      const std::vector<RowVectorPtr>& probeInput,
-      const std::vector<RowVectorPtr>& buildInput,
       const core::PlanNodePtr& plan);
 
   int32_t randInt(int32_t min, int32_t max) {
     return boost::random::uniform_int_distribution<int32_t>(min, max)(rng_);
   }
 
-  static inline const std::string kHiveConnectorId = "test-hive";
-
-  static std::shared_ptr<connector::ConnectorSplit> makeSplit(
-      const std::string& filePath);
-
   FuzzerGenerator rng_;
   size_t currentSeed_{0};
 
   std::shared_ptr<memory::MemoryPool> rootPool_{
-      memory::memoryManager()->addRootPool()};
-  std::shared_ptr<memory::MemoryPool> pool_{rootPool_->addLeafChild("leaf")};
+      memory::memoryManager()->addRootPool(
+          "joinFuzzer",
+          memory::kMaxMemory,
+          memory::MemoryReclaimer::create())};
+  std::shared_ptr<memory::MemoryPool> pool_{rootPool_->addLeafChild(
+      "joinFuzzerLeaf",
+      true,
+      exec::MemoryReclaimer::create())};
+  std::shared_ptr<memory::MemoryPool> writerPool_{rootPool_->addAggregateChild(
+      "joinFuzzerWriter",
+      exec::MemoryReclaimer::create())};
+
   VectorFuzzer vectorFuzzer_;
+
+  // The following fields are reset cross test iterations.
+  core::JoinType joinType_;
```

Comment:
This is a very long list... Will it be difficult to maintain? Is there some other way to organize the code for easier maintenance?

Reply:
Put it in a separate class.